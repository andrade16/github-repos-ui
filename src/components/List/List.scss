.list-container {
  list-style-type: none;
  padding: 0;
  margin-top: 0;
}
