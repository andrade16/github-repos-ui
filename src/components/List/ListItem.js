import React from "react";
import Avatar from "../Avatar/Avatar";
import "./ListItem.scss";

const ListItem = ({ title, subtitle, content, avatarSrc }) => {
  return (
    <li className="list-item">
      <div className="titles-container">
        <div className="avatar-title-container">
          <Avatar width={30} height={30} src={avatarSrc} />
          <div className="title">{title}</div>
        </div>
        <div className="subtitle">{subtitle}</div>
      </div>
      <div className="content-container">{content}</div>
    </li>
  );
};

export default ListItem;
