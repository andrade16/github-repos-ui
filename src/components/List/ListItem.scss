.list-item {
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
  margin: 5px;

  .titles-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 4px;

    .avatar-title-container {
      display: flex;
      flex-direction: row;
      align-items: center;
    }

    .title,
    .subtitle {
      margin: 5px;
    }

    .title {
      font-size: 14px;
      font-weight: bold;
    }

    .subtitle {
      font-size: 12px;
    }
  }

  .content-container {
    padding: 4px;
    font-size: 12px;
  }
}
