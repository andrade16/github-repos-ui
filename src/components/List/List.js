import React from "react";
import ListItem from "./ListItem";
import { formatDate } from "../../utils/helpers";
import "./List.scss";

const List = ({ data }) => {
  let listContent = null;
  if (data.length) {
    listContent = data.map((commit) => (
      <ListItem
        key={commit.sha}
        title={commit.author.login}
        subtitle={`committed ${formatDate(commit.commit.committer.date)}`}
        content={commit.commit.message}
        avatarSrc={commit.author.avatar_url}
      />
    ));
  }

  return <ul className="list-container">{listContent}</ul>;
};

export default List;
