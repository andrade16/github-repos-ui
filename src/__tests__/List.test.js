import React from "react";
import renderer from "react-test-renderer";
import List from "../components/List/List";

const mockCommitObjs = [
  {
    sha: "950841c25ea005f30050cdac545f87933cea038a",
    author: {
      login: "nhcarrigan",
      avatar_url: "https://avatars.githubusercontent.com/u/63889819?v=4",
    },
    commit: {
      committer: { date: "2021-02-18T16:04:03Z" },
      message: "test",
    },
  },
  {
    sha: "950841c25ea005f30050cdac545f87933fffffff",
    author: {
      login: "wla3ww",
      avatar_url: "https://avatars.githubusercontent.com/u/63889819?v=5",
    },
    commit: {
      committer: { date: "2021-02-18T16:04:03Z" },
      message: "test",
    },
  },
];

describe("List", () => {
  it("Renders List correctly", () => {
    const tree = renderer.create(<List data={mockCommitObjs} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
