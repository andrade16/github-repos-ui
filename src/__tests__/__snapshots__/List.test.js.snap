// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Renders List correctly 1`] = `
<ul
  className="list-container"
>
  <li
    className="list-item"
  >
    <div
      className="titles-container"
    >
      <div
        className="avatar-title-container"
      >
        <div
          className="avatar"
        >
          <img
            className="avatar-image"
            src="https://avatars.githubusercontent.com/u/63889819?v=4"
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </div>
        <div
          className="title"
        >
          nhcarrigan
        </div>
      </div>
      <div
        className="subtitle"
      >
        committed 12 hours ago
      </div>
    </div>
    <div
      className="content-container"
    >
      test
    </div>
  </li>
  <li
    className="list-item"
  >
    <div
      className="titles-container"
    >
      <div
        className="avatar-title-container"
      >
        <div
          className="avatar"
        >
          <img
            className="avatar-image"
            src="https://avatars.githubusercontent.com/u/63889819?v=5"
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </div>
        <div
          className="title"
        >
          wla3ww
        </div>
      </div>
      <div
        className="subtitle"
      >
        committed 12 hours ago
      </div>
    </div>
    <div
      className="content-container"
    >
      test
    </div>
  </li>
</ul>
`;
